python/core/additions/qgsfunction.py

```diff
@@ -118,7 +118,7 @@ def register_function(
     :param args: DEPRECATED since QGIS 3.32.  Use ``params_as_list`` if you want to pass parameters as a list.
     :param group: the expression group in which the function should be added
     :param usesgeometry: Defines if this expression requires the geometry. By default False.
-    :param referenced_columns: An array of field names on which this expression works. By default ``[QgsFeatureRequest.ALL_ATTRIBUTES]``. Can be set to None for slightly faster evaluation.
+    :param referenced_columns: An array of field names on which this expression works. By default ``[QgsFeatureRequest.ALL_ATTRIBUTES]``. Can be set to an empty list for slightly faster evaluation.
```

Comment:
```suggestion
    :param referenced_columns: An array of names of fields on which this expression works. By default ``[QgsFeatureRequest.ALL_ATTRIBUTES]``. Specifying a subset of fields or an empty list will result in a faster execution.
```

I think this can make a big difference in some scenarios, so removing the "slightly" sounds appropriate. The rest is a proposal for a reformulation.

Reply:
I agree, we could even add that such performance improvement depends on the provider, but that's probably too low-level-info for end users.